__kommandir__/bin/adept_openstack.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python2
+
+import sys
+import os
+import os.path
+# ref: https://github.com/openstack/os-client-config
+import os_client_config
+
+# FIXME: Make more generic, so it can provision peons also.  Needs
+#        to accept 'volume & size' along with 'image' arguments.
+#        consider switching to argparse.
+
+IMAGE = 'CentOS-Cloud-7'
+FLAVOR='m1.medium',
+ALIASES = {'openstack_destroy.py': 'destroy',
+           'openstack_discover_create.py': 'discover_create'}
+BASENAME = os.path.basename(sys.argv[0])
+if BASENAME not in ALIASES.keys():
+    raise RuntimeError("Script must be called as one of\n%s"
+                       % ALIASES.keys())
+
+
+MODE = ALIASES[BASENAME]
+if MODE == 'destroy':
+    if len(sys.argv) < 2:
+        raise RuntimeError("Must pass server name as first parameter")
+elif MODE == 'discover_create':
+    if len(sys.argv) < 3:
+        raise RuntimeError("Must pass server name, and one or more"
+                           " paths to public ssh key files as parameters")
+    PUB_KEY_FILES = sys.argv[2:]
+else:
+    raise RuntimeError("Unknown operation mode %s", MODE)
+NAME = sys.argv[1]
+
+
+# Assumed script running under an exclusive lock to prevent TOCTOU race
+session = os_client_config.make_rest_client('compute')
+response = session.get('/servers')
+servers_json = response.json()['servers']
+matched_server = [s for s in servers_json
+                  if s['name'] == NAME]
+
+
+if len(matched_server) > 1:
+    raise ValueError("Found more than one server named %s" % NAME)
+elif len(matched_server) == 1:
+    matched_server = matched_server[0]
+    response = session.get('/servers/%s' % matched_server['id'])
+    matched_server = response.json()['server']
+    MODE='discover'
```

Comment:
Aye, you caught me.   I'm not very fond of using ``MODE`` anyway, and the same-file w/ symlinks maybe overrated.  Okay, I'll add this to the chopping-block as well.  Thanks.